components/org.wso2.carbon.appmgt.hostobjects/src/main/java/org/wso2/carbon/appmgt/hostobjects/ThemeManagerHostObject.java

```diff
@@ -0,0 +1,437 @@
+        StringBuilder path = new StringBuilder(getStoreThemePath());
+        path.append(File.separator).append(tenant);
+        return path.toString();
+    }
+
+    /**
+     * Construct and return the custom tenant theme extensions path.
+     * [/repository/deployment/server/jaggeryapps/store/themes/<tenantDomain>/extensions/assets]
+     *
+     * @return path
+     */
+    private static String getExtThemePath(String tenantDomain) {
+        StringBuilder path = new StringBuilder(getTenantThemePath(tenantDomain));
+        path.append(File.separator).append("extensions").append(File.separator)
+                .append("assets");
+        return path.toString();
+    }
+
+    /**
+     * Construct and return the custom tenant theme path.
+     *
+     * @param tenantDomain Tenant Domain
+     * @param themeType    Theme Type (Default ,<assetType> e.g webapp)
+     * @return path
+     */
+    private static String getCustomThemePath(String tenantDomain, String themeType) {
+        StringBuilder path = new StringBuilder(getTenantThemePath(tenantDomain));
+        if (DEFAULT.equals(themeType)) {
+            path.append(File.separator).append("themes").append(File.separator)
+                    .append("custom");
+        } else {
+            path.append(File.separator).append("extensions").append(File.separator)
+                    .append("assets").append(File.separator).append(themeType).append(File.separator).append("themes")
+                    .append(File.separator).append("custom");
+        }
+        return path.toString();
+    }
+
+
+    @Override
+    public String getClassName() {
+        return "ThemeManager";
+    }
+
+    public static Scriptable jsConstructor(Context cx, Object[] args, Function ctorObj, boolean inNewExpr)
+            throws ScriptException, AppManagementException {
+        return new ThemeManagerHostObject();
+    }
+
+    private static void handleException(String msg) throws AppManagementException {
+        log.error(msg);
+        throw new AppManagementException(msg);
+    }
+
+    private static void handleException(String msg, Throwable t) throws AppManagementException {
+        log.error(msg, t);
+        throw new AppManagementException(msg, t);
+    }
+
+    /**
+     * Add the custom theme for given tenant based on the theme type(eg. default,webapp,site).
+     *
+     * @param cx
+     * @param thisObj
+     * @param args
+     * @param funObj
+     * @return true if success
+     * @throws AppManagementException
+     */
+    public static boolean jsFunction_addCustomTheme(Context cx, Scriptable thisObj, Object[] args, Function funObj)
+            throws AppManagementException {
+        if (args == null || args.length != 4) {
+            handleException("Invalid input parameters for addTenantTheme");
+        }
+
+        FileHostObject uploadFile = (FileHostObject) args[0];
+        String tenant = (String) args[1];
+        String themeType = (String) args[2];
```

Comment:
Get the themetype(=asset types) from backend api rather configuring in site.json